src/main/java/echoserver/SocketCloseException.java

```diff
@@ -2,7 +2,7 @@
 
 class SocketCloseException extends RuntimeException {
```

Comment:
I don't fully understand why do you need both SockerCloseExceptioon and SocketOpenException if they aren't really any different.